### PR TITLE
Update DG: Implementation details 

### PR DESCRIPTION
**Pull request by enhao25**
Add implementation details for other student command. (Do not intend to do diagrams for them)
Add sequence diagram for add student feature.
Update session's DG to be in line with the student format.
Add get monthly fee feature to DG.
Remove "proposed undo feature"
Update Value proposition.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -270,91 +293,48 @@ The following sequence diagram shows how deleting a session works:
...
-
-The following activity diagram summarizes what happens when a user executes a new command:
-
-![CommitActivityDiagram](images/CommitActivityDiagram.png)
-
-#### Design consideration:
-
-##### Aspect: How undo & redo executes
+#### Implementation
+The creation of a session is facilitated by `GetMonthlyFeeCommand` and it extends `Command`. The method,
```
Comment (JonahhGohh) [inline docs/DeveloperGuide.md]: Is this part an err? Doesn't seem to describe the GetMonthlyFeeCommand.

**Review by JonahhGohh — COMMENTED** [review]
Yo thanks for working on the DG!
I went to check the completed DG from gabriel's grp and some other grps and noticed that they did not list out every single command as a feature. Like all these add/edit/delete student commands were not present.
eg: https://ay1920s1-cs2103t-f12-2.github.io/main/DeveloperGuide.html

I think our DG needs some structure rework, if not might be bloated with a lot of similar information. What do you think?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (enhao25) [reply docs/DeveloperGuide.md]: Ah right! Nice catch. O.o hmmm in this case, we might not need to include the all the different commands for students inside the DG then. I tink the rest of the sections is good, just that the implementation sections could be restructured, but not sure how we should do it also tou. Should I remove those other student's command section first?

**Review by nowknowing — APPROVED** [review]
Nice implementation and explanation with FeeUtil,
LGTM!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -270,91 +293,48 @@ The following sequence diagram shows how deleting a session works:
 
 It shares the same design considerations as what is mentioned in Add Session Feature.
 
+### Calculating Monthly Fee Feature
+The monthly `fee` feature allows user to quickly calculate the amount of money they would have received
```
Comment (nowknowing) [inline docs/DeveloperGuide.md]: Should receive/would receive.

"Would have received" may wrongly suggest to reader that only past sessions would be included in fee calculation

**Review by nowknowing — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (enhao25) [reply docs/DeveloperGuide.md]: O.o right~! Have updated accordingly. Thanks 👍🏻

**Review by nowknowing — APPROVED** [review]
LGTM!

**Review by JonahhGohh — APPROVED** [review]
LGTM

**Pull request merged**